enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+---
+title: aws-efa-networking
+authors:
+  - "@JoelSpeed"
+reviewers:
+  - "@elmiko"
+  - "@deads2k"
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+  - "@deads2k"
+creation-date: 2021-12-15
+last-updated: 2021-12-15
+tracking-link:
+  - https://issues.redhat.com/browse/OCPCLOUD-1353
+---
+
+# AWS EFA Networking
+
+## Summary
+
+AWS Elastic-Fabric-Adapter (EFA) is a feature within EC2 that allows for high throughput network adapters to be
+attached to certain instance types. With the presence of these network adapters and some additional drivers,
+workloads can take advantage of greatly improved network performance between nodes within the same AWS availability
+zone.
+
+## Motivation
+
+We have customers already using this feature of EC2 within OpenShift UPI clusters. They have taken on the responsibility
+of making sure the prerequisites are fulfilled and have manually created instances with the attached adapters.
+However, there is a desire to enable this feature to be used in conjunction with existing autoscaling features within
+OpenShift. To enable this, we must provide support for the feature within the Machine API.
+
+### Goals
+
+- Enable worker Machines to attach EFA network interfaces as their primary network interface
+- Enable autoscaling of Machines that leverage high performance network interfaces
```

Comment:
Are we _sure_ that EFA support is the only feature that blocks autoscaling?

Because it looks like they are probably going to want to [put the HPC nodes into a "placement group"](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-instances) to cause them to all be deployed on physically-close hardware, and it looks like `AWSMachineProviderConfig` does not allow you to specify that. (This seems to be more precise than just `machinev1beta1.Placement`.)

You _can_ set a placement group post-creation/installation... But that involves shutting down and migrating the VM, which kills most of the benefit of autoscaling.

Reply:
@romfreiman Do we know if the customer driving this currently relies on placement groups for their UPI based approach?

Reply:
@tkatarki  any idea?

Also, Note from the link above:
It is not an absolute requirement to launch your EFA-enabled instances into a cluster placement group. However, we do recommend running your EFA-enabled instances in a cluster placement group as it launches the instances into a low-latency group in a single Availability Zone.